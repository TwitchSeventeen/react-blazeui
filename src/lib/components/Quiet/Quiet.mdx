---
name: Quiet
---

import { Playground, PropsTable } from 'docz'
import Quiet from './index';

# Quiet

This adds a quiet emphasis to text. It can be used inline (ie it's not a block level element).

## Properties
<PropsTable of={Quiet} />

## Basic usage
<Playground>
    <Quiet>
        other times you want to whisper
    </Quiet>
    <br />
    Regular text
</Playground>