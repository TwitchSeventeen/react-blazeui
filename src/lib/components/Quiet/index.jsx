import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Quiet(props) {
  const {
    children, className, dataName, ...other
  } = props;
  const classes = classNames('u-text--quiet', className);
  return (
    <span
      {...other}
      className={classes}
      data-name={dataName}
    >
      { children }
    </span>
  );
}

Quiet.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * Optional data-name attribute
   * (use for testing/selection)
   */
  dataName: PropTypes.string,
};

Quiet.defaultProps = {
  children: null,
  className: null,
  dataName: 'quiet',
};

export default Quiet;
