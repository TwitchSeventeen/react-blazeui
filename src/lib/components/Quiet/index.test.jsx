import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Quiet from './index';

describe('Quiet', () => {
  afterEach(cleanup);

  it('should be defined', () => {
    expect(Quiet).toBeDefined();
  });

  it('renders correctly', () => {
    const { asFragment } = render(
      <Quiet>
        lorem ipsum dolor
      </Quiet>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
