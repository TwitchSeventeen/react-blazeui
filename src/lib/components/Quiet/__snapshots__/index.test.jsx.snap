// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quiet renders correctly 1`] = `
<DocumentFragment>
  <span
    class="u-text--quiet"
    data-name="quiet"
  >
    lorem ipsum dolor
  </span>
</DocumentFragment>
`;
